## Atliq Mart: Sales and Promotion Insights

**Delving deep into Atliq Marts data to derive insights hidden in the data, the goal 
of this analysis is to discover the most effective and profitable promotion methods, in order to ascertain 
those that are working and ones that are not and help Atliq mart make informed decisions.**

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
import warnings
import seaborn as sns

warnings.filterwarnings("ignore")

In [3]:
data = pd.read_csv(".\\dataset\\fact_events.csv")
data.head()

event_id store_id  campaign_id product_code  base_price    promo_type  \
0   7f650b  STCBE-2  CAMP_SAN_01          P11         190       50% OFF   
1   a21f91  STBLR-8  CAMP_DIW_01          P03         156       25% OFF   
2   78bc80  STVJD-0  CAMP_SAN_01          P07         300         BOGOF   
3   a1503f  STCBE-1  CAMP_DIW_01          P15        3000  500 Cashback   
4   1091cf  STBLR-6  CAMP_DIW_01          P05          55       25% OFF   

   quantity_sold(before_promo)  quantity_sold(after_promo)  
0                           34                          52  
1                          393                         322  
2                           22                          85  
3                          329                        1000  
4                          108                          93

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500 entries, 0 to 1499
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   event_id                     1500 non-null   object
 1   store_id                     1500 non-null   object
 2   campaign_id                  1500 non-null   object
 3   product_code                 1500 non-null   object
 4   base_price                   1500 non-null   int64 
 5   promo_type                   1500 non-null   object
 6   quantity_sold(before_promo)  1500 non-null   int64 
 7   quantity_sold(after_promo)   1500 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 93.9+ KB


In [5]:
data.dtypes

event_id                       object
store_id                       object
campaign_id                    object
product_code                   object
base_price                      int64
promo_type                     object
quantity_sold(before_promo)     int64
quantity_sold(after_promo)      int64
dtype: object

In [6]:
data.rename({"quantity_sold(before_promo)":"pre_promo_sales", "quantity_sold(after_promo)":"post_promo_sales"}, axis=1, inplace = True)
data.head()

event_id store_id  campaign_id product_code  base_price    promo_type  \
0   7f650b  STCBE-2  CAMP_SAN_01          P11         190       50% OFF   
1   a21f91  STBLR-8  CAMP_DIW_01          P03         156       25% OFF   
2   78bc80  STVJD-0  CAMP_SAN_01          P07         300         BOGOF   
3   a1503f  STCBE-1  CAMP_DIW_01          P15        3000  500 Cashback   
4   1091cf  STBLR-6  CAMP_DIW_01          P05          55       25% OFF   

   pre_promo_sales  post_promo_sales  
0               34                52  
1              393               322  
2               22                85  
3              329              1000  
4              108                93

In [7]:
data['Incr_sold_units'] = data['post_promo_sales'] - data['pre_promo_sales']
data['Incr_revenue'] = data['Incr_sold_units'] * data['base_price']
data['perc_incr_units'] = (data['Incr_sold_units']/data['pre_promo_sales']) * 100
data['perc_incr_revenue'] = (data['Incr_revenue']/(data['base_price'] *data['pre_promo_sales'])) * 100

In [8]:
data = data.round(2)

In [9]:
data.head()

event_id store_id  campaign_id product_code  base_price    promo_type  \
0   7f650b  STCBE-2  CAMP_SAN_01          P11         190       50% OFF   
1   a21f91  STBLR-8  CAMP_DIW_01          P03         156       25% OFF   
2   78bc80  STVJD-0  CAMP_SAN_01          P07         300         BOGOF   
3   a1503f  STCBE-1  CAMP_DIW_01          P15        3000  500 Cashback   
4   1091cf  STBLR-6  CAMP_DIW_01          P05          55       25% OFF   

   pre_promo_sales  post_promo_sales  Incr_sold_units  Incr_revenue  \
0               34                52               18          3420   
1              393               322              -71        -11076   
2               22                85               63         18900   
3              329              1000              671       2013000   
4              108                93              -15          -825   

   perc_incr_units  perc_incr_revenue  
0            52.94              52.94  
1           -18.07             -18.07  
2           286.36             286.36  
3           203.95             203.95  
4           -13.89             -13.89

In [10]:
data['revenue'] =  data['post_promo_sales'] * data['base_price']

In [11]:
stores_grp = data.groupby(['store_id'])
stores_rev = pd.DataFrame(stores_grp['Incr_revenue'].sum()).reset_index()
stores_rev.head()

store_id  Incr_revenue
0  STBLR-0       6158906
1  STBLR-1       3602620
2  STBLR-2       4078910
3  STBLR-3       5209263
4  STBLR-4       5024831

In [12]:
stores_isu = pd.DataFrame(stores_grp['perc_incr_revenue'].mean().round(2)).reset_index()
stores_isu.head()

store_id  perc_incr_revenue
0  STBLR-0             125.68
1  STBLR-1              78.75
2  STBLR-2             109.97
3  STBLR-3             107.04
4  STBLR-4             110.61

In [13]:
stores_units = pd.DataFrame(stores_grp['Incr_sold_units'].sum()).reset_index()
stores_units.head()

store_id  Incr_sold_units
0  STBLR-0             6487
1  STBLR-1             3770
2  STBLR-2             4781
3  STBLR-3             5777
4  STBLR-4             5644

In [14]:
stores_rev_avg = pd.DataFrame(stores_grp['Incr_revenue'].mean().round(2)).reset_index()
stores_rev_avg.head()

store_id  Incr_revenue
0  STBLR-0     205296.87
1  STBLR-1     120087.33
2  STBLR-2     135963.67
3  STBLR-3     173642.10
4  STBLR-4     167494.37

In [15]:
stores_tot_sales = pd.DataFrame(stores_grp['revenue'].sum().round(2)).reset_index()
stores_tot_sales.head()

store_id  revenue
0  STBLR-0  9559971
1  STBLR-1  6781882
2  STBLR-2  7096826
3  STBLR-3  8569975
4  STBLR-4  8296143

<br><br>

<u><h3>Store Performance Analysis</h3><u>

**Top 5 Stores By Incremental Revenue**

In [16]:
store_grp = data.groupby(['store_id', 'promo_type'])

top_perf_stores = pd.DataFrame(store_grp['Incr_revenue'].sum())


store_total_revenue = data.groupby('store_id')['Incr_revenue'].sum()


top_perf_stores = pd.merge(top_perf_stores, store_total_revenue, left_index=True, right_index=True, suffixes=('_promo', '_total'))


top_perf_stores['Incr_revenue_percentage'] = (top_perf_stores['Incr_revenue_promo'] / top_perf_stores['Incr_revenue_total']) * 100

top_perf_stores = pd.DataFrame(top_perf_stores.sort_values(by='Incr_revenue_total', ascending=False))

In [17]:
top_perf_stores[:24]

Incr_revenue_promo  Incr_revenue_total  \
store_id promo_type                                             
STMYS-1  50% OFF                    25583             6446961   
         500 Cashback             3900000             6446961   
         25% OFF                   -35626             6446961   
         33% OFF                   596324             6446961   
         BOGOF                    1960680             6446961   
STCHE-4  25% OFF                   -23812             6317711   
         33% OFF                   438600             6317711   
         50% OFF                    19713             6317711   
         500 Cashback             3918000             6317711   
         BOGOF                    1965210             6317711   
STBLR-0  33% OFF                   331616             6158906   
         25% OFF                   -27723             6158906   
         BOGOF                    1906620             6158906   
         500 Cashback             3930000             6158906   
         50% OFF                    18393             6158906   
STBLR-7  BOGOF                    2007930             6124481   
         500 Cashback             3594000             6124481   
         50% OFF                    28788             6124481   
         33% OFF                   520300             6124481   
         25% OFF                   -26537             6124481   
STBLR-6  500 Cashback             3498000             6008349   
         50% OFF                    26309             6008349   
         33% OFF                   514624             6008349   
         25% OFF                   -27584             6008349   

                       Incr_revenue_percentage  
store_id promo_type                             
STMYS-1  50% OFF                      0.396823  
         500 Cashback                60.493619  
         25% OFF                     -0.552601  
         33% OFF                      9.249691  
         BOGOF                       30.412469  
STCHE-4  25% OFF                     -0.376909  
         33% OFF                      6.942388  
         50% OFF                      0.312028  
         500 Cashback                62.016132  
         BOGOF                       31.106361  
STBLR-0  33% OFF                      5.384333  
         25% OFF                     -0.450129  
         BOGOF                       30.957121  
         500 Cashback                63.810034  
         50% OFF                      0.298641  
STBLR-7  BOGOF                       32.785309  
         500 Cashback                58.682523  
         50% OFF                      0.470048  
         33% OFF                      8.495414  
         25% OFF                     -0.433294  
STBLR-6  500 Cashback                58.218988  
         50% OFF                      0.437874  
         33% OFF                      8.565148  
         25% OFF                     -0.459095

<p><p>

**Bottom 5 Stores By Incremental Revenue**

In [18]:
top_perf_stores.tail(25)

Incr_revenue_promo  Incr_revenue_total  \
store_id promo_type                                             
STMLR-1  BOGOF                     880630             2465544   
         500 Cashback             1419000             2465544   
         50% OFF                     8526             2465544   
         33% OFF                   172860             2465544   
         25% OFF                   -15472             2465544   
STTRV-0  BOGOF                     865590             2416197   
         500 Cashback             1347000             2416197   
         25% OFF                    -6156             2416197   
         33% OFF                   202444             2416197   
         50% OFF                     7319             2416197   
STVSK-3  BOGOF                     684210             2333453   
         500 Cashback             1512000             2333453   
         50% OFF                     4416             2333453   
         33% OFF                   151704             2333453   
         25% OFF                   -18877             2333453   
STTRV-1  BOGOF                     815390             2249863   
         500 Cashback             1251000             2249863   
         33% OFF                   181288             2249863   
         50% OFF                     8254             2249863   
         25% OFF                    -6069             2249863   
STMLR-0  25% OFF                   -14930             1753077   
         33% OFF                   167528             1753077   
         50% OFF                     8719             1753077   
         500 Cashback             1044000             1753077   
         BOGOF                     547760             1753077   

                       Incr_revenue_percentage  
store_id promo_type                             
STMLR-1  BOGOF                       35.717472  
         500 Cashback                57.553222  
         50% OFF                      0.345806  
         33% OFF                      7.011029  
         25% OFF                     -0.627529  
STTRV-0  BOGOF                       35.824480  
         500 Cashback                55.748766  
         25% OFF                     -0.254781  
         33% OFF                      8.378621  
         50% OFF                      0.302914  
STVSK-3  BOGOF                       29.321782  
         500 Cashback                64.796677  
         50% OFF                      0.189247  
         33% OFF                      6.501267  
         25% OFF                     -0.808973  
STTRV-1  BOGOF                       36.241762  
         500 Cashback                55.603386  
         33% OFF                      8.057735  
         50% OFF                      0.366867  
         25% OFF                     -0.269750  
STMLR-0  25% OFF                     -0.851645  
         33% OFF                      9.556226  
         50% OFF                      0.497354  
         500 Cashback                59.552433  
         BOGOF                       31.245633

<p><p>

In [19]:
store_grp = data.groupby(['store_id', 'promo_type'])

In [20]:
store_data = pd.read_csv(".\\dataset\\dim_stores.csv")
store_data.head()

store_id           city
0  STTRV-0     Trivandrum
1  STMDU-3        Madurai
2  STHYD-6      Hyderabad
3  STVSK-1  Visakhapatnam
4  STCBE-3     Coimbatore

In [21]:
store_merged = data.merge(store_data, on='store_id')
store_merged = store_merged.round(2)
store_merged.head()

event_id store_id  campaign_id product_code  base_price    promo_type  \
0   7f650b  STCBE-2  CAMP_SAN_01          P11         190       50% OFF   
1   ca7298  STCBE-2  CAMP_SAN_01          P15        3000  500 Cashback   
2   35fb5b  STCBE-2  CAMP_DIW_01          P13         350         BOGOF   
3   0f1be6  STCBE-2  CAMP_DIW_01          P12          62       50% OFF   
4   3df185  STCBE-2  CAMP_DIW_01          P11         190       50% OFF   

   pre_promo_sales  post_promo_sales  Incr_sold_units  Incr_revenue  \
0               34                52               18          3420   
1               85               228              143        429000   
2               50               190              140         49000   
3               89               129               40          2480   
4               40                62               22          4180   

   perc_incr_units  perc_incr_revenue  revenue        city  
0            52.94              52.94     9880  Coimbatore  
1           168.24             168.24   684000  Coimbatore  
2           280.00             280.00    66500  Coimbatore  
3            44.94              44.94     7998  Coimbatore  
4            55.00              55.00    11780  Coimbatore

<p><p>

**Stores Overview**

In [22]:
stores_overview = pd.DataFrame(stores_grp['Incr_revenue'].sum().reset_index())

stores_isu.rename(columns={'Incr_revenue': 'Incr_revenue_isu', 'perc_incr_revenue': 'perc_incr_revenue_isu', 'Incr_sold_units': 'Incr_sold_units_isu'}, inplace=True)
stores_units.rename(columns={'Incr_revenue': 'Incr_revenue_units', 'perc_incr_revenue': 'perc_incr_revenue_units', 'Incr_sold_units': 'Incr_sold_units_units'}, inplace=True)
stores_rev_avg.rename(columns={'Incr_revenue': 'Incr_revenue_rev_avg', 'perc_incr_revenue': 'perc_incr_revenue_rev_avg', 'Incr_sold_units': 'Incr_sold_units_rev_avg'}, inplace=True)


In [23]:
stores_overview = pd.merge(stores_overview, stores_isu, on='store_id', how='left')
stores_overview = pd.merge(stores_overview, stores_units, on='store_id', how='left')
stores_overview = pd.merge(stores_overview, stores_rev_avg, on='store_id', how='left')
stores_overview = pd.merge(stores_overview, stores_tot_sales, on='store_id', how='left')

stores_overview['%revenue_increment'] = (stores_overview['Incr_revenue']/stores_overview['revenue']) * 100
stores_overview = stores_overview.round(2)

In [24]:
stores_overview.sort_values(by='Incr_revenue', ascending=False)

store_id  Incr_revenue  perc_incr_revenue_isu  Incr_sold_units_units  \
38  STMYS-1       6446961                 131.91                   6862   
19  STCHE-4       6317711                 128.80                   6453   
0   STBLR-0       6158906                 125.68                   6487   
7   STBLR-7       6124481                 133.74                   6859   
6   STBLR-6       6008349                 130.61                   6573   
22  STCHE-7       5999008                 132.18                   6560   
40  STMYS-3       5721938                 120.42                   5284   
18  STCHE-3       5707253                 127.63                   6091   
21  STCHE-6       5324877                 106.59                   5641   
3   STBLR-3       5209263                 107.04                   5777   
8   STBLR-8       5110214                 107.15                   5475   
23  STHYD-0       5086448                 115.90                   5705   
4   STBLR-4       5024831                 110.61                   5644   
25  STHYD-2       4961263                 106.66                   5773   
20  STCHE-5       4856775                 104.40                   5243   
30  STMDU-0       4852087                 126.44                   4724   
5   STBLR-5       4840874                 109.97                   5363   
29  STHYD-6       4763361                 109.89                   5445   
9   STBLR-9       4605894                 113.09                   5241   
15  STCHE-0       4602167                 110.23                   5254   
28  STHYD-5       4549480                 111.83                   5515   
17  STCHE-2       4522829                 105.53                   5245   
27  STHYD-4       4448381                 111.88                   5396   
12  STCBE-2       4145314                 131.65                   4960   
2   STBLR-2       4078910                 109.97                   4781   
10  STCBE-0       4024491                 117.67                   4807   
33  STMDU-3       3931241                 104.23                   3950   
39  STMYS-2       3857145                  75.75                   3411   
32  STMDU-2       3813777                 103.02                   4058   
11  STCBE-1       3803491                 100.66                   3885   
26  STHYD-3       3799948                  81.54                   4015   
31  STMDU-1       3753727                 110.58                   3979   
1   STBLR-1       3602620                  78.75                   3770   
46  STVSK-1       3481374                 107.67                   4003   
13  STCBE-3       3415604                 114.18                   4171   
45  STVSK-0       3408675                 115.57                   4116   
24  STHYD-1       3231466                  69.19                   3187   
16  STCHE-1       3170382                  67.97                   3281   
47  STVSK-2       3062223                 115.08                   3944   
37  STMYS-0       2829433                  74.57                   3344   
14  STCBE-4       2779806                  75.56                   2927   
43  STVJD-0       2733934                 112.56                   3046   
44  STVJD-1       2684369                 107.50                   2763   
49  STVSK-4       2568489                  72.06                   2469   
36  STMLR-2       2521058                 103.27                   2664   
35  STMLR-1       2465544                 109.72                   2784   
41  STTRV-0       2416197                 110.55                   2733   
48  STVSK-3       2333453                  69.07                   2209   
42  STTRV-1       2249863                 111.24                   2604   
34  STMLR-0       1753077                  74.55                   1952   

    Incr_revenue_rev_avg  revenue  %revenue_increment  
38             214898.70  9986244               64.56  
19             210590.37  9874571               63.98  
0        

<p><p>

**Overview of Top Stores**

In [25]:
stores_overview.sort_values(by='Incr_revenue',ascending=False,inplace=True)
stores_overview.head()

store_id  Incr_revenue  perc_incr_revenue_isu  Incr_sold_units_units  \
38  STMYS-1       6446961                 131.91                   6862   
19  STCHE-4       6317711                 128.80                   6453   
0   STBLR-0       6158906                 125.68                   6487   
7   STBLR-7       6124481                 133.74                   6859   
6   STBLR-6       6008349                 130.61                   6573   

    Incr_revenue_rev_avg  revenue  %revenue_increment  
38             214898.70  9986244               64.56  
19             210590.37  9874571               63.98  
0              205296.87  9559971               64.42  
7              204149.37  9479717               64.61  
6              200278.30  9412361               63.83

**Overview of Bottom Stores**

In [26]:
stores_overview.tail()

store_id  Incr_revenue  perc_incr_revenue_isu  Incr_sold_units_units  \
35  STMLR-1       2465544                 109.72                   2784   
41  STTRV-0       2416197                 110.55                   2733   
48  STVSK-3       2333453                  69.07                   2209   
42  STTRV-1       2249863                 111.24                   2604   
34  STMLR-0       1753077                  74.55                   1952   

    Incr_revenue_rev_avg  revenue  %revenue_increment  
35              82184.80  4111794               59.96  
41              80539.90  4099803               58.93  
48              77781.77  4625957               50.44  
42              74995.43  3768043               59.71  
34              58435.90  3383973               51.81

<p><p>

**Performance of Stores by City**

In [27]:
store_city_grp = store_merged.groupby(['city','store_id'])

store_city = pd.DataFrame(store_city_grp['Incr_revenue'].sum())

In [28]:
store_city

Incr_revenue
city          store_id              
Bengaluru     STBLR-0        6158906
              STBLR-1        3602620
              STBLR-2        4078910
              STBLR-3        5209263
              STBLR-4        5024831
              STBLR-5        4840874
              STBLR-6        6008349
              STBLR-7        6124481
              STBLR-8        5110214
              STBLR-9        4605894
Chennai       STCHE-0        4602167
              STCHE-1        3170382
              STCHE-2        4522829
              STCHE-3        5707253
              STCHE-4        6317711
              STCHE-5        4856775
              STCHE-6        5324877
              STCHE-7        5999008
Coimbatore    STCBE-0        4024491
              STCBE-1        3803491
              STCBE-2        4145314
              STCBE-3        3415604
              STCBE-4        2779806
Hyderabad     STHYD-0        5086448
              STHYD-1        3231466
              STHYD-2        4961263
              STHYD-3        3799948
              STHYD-4        4448381
              STHYD-5        4549480
              STHYD-6        4763361
Madurai       STMDU-0        4852087
              STMDU-1        3753727
              STMDU-2        3813777
              STMDU-3        3931241
Mangalore     STMLR-0        1753077
              STMLR-1        2465544
              STMLR-2        2521058
Mysuru        STMYS-0        2829433
              STMYS-1        6446961
              STMYS-2        3857145
              STMYS-3        5721938
Trivandrum    STTRV-0        2416197
              STTRV-1        2249863
Vijayawada    STVJD-0        2733934
              STVJD-1        2684369
Visakhapatnam STVSK-0        3408675
              STVSK-1        3481374
              STVSK-2        3062223
              STVSK-3        2333453
              STVSK-4        2568489

In [29]:
city_grp = store_merged.groupby(['city'])
city_tot_sales = pd.DataFrame(city_grp['revenue'].sum().reset_index())
city_tot_sales.head()

city   revenue
0   Bengaluru  83707621
1     Chennai  66787456
2  Coimbatore  30404660
3   Hyderabad  53519028
4     Madurai  26682384

**Incremental Revenue by City**

In [30]:
city_rev_sum = pd.DataFrame(city_grp['Incr_revenue'].sum().reset_index())

city_rev_sum

city  Incr_revenue
0      Bengaluru      50764342
1        Chennai      40501002
2     Coimbatore      18168706
3      Hyderabad      30840347
4        Madurai      16350832
5      Mangalore       6739679
6         Mysuru      18855477
7     Trivandrum       4666060
8     Vijayawada       5418303
9  Visakhapatnam      14854214

**City Overview**

In [31]:
city_overview = pd.merge(city_tot_sales, city_rev_sum,on='city')

city_overview['%revenue_increment'] = (city_overview['Incr_revenue']/city_overview['revenue']) * 100

city_overview = city_overview.round(2)

city_overview

city   revenue  Incr_revenue  %revenue_increment
0      Bengaluru  83707621      50764342               60.64
1        Chennai  66787456      40501002               60.64
2     Coimbatore  30404660      18168706               59.76
3      Hyderabad  53519028      30840347               57.63
4        Madurai  26682384      16350832               61.28
5      Mangalore  11774495       6739679               57.24
6         Mysuru  31801764      18855477               59.29
7     Trivandrum   7867846       4666060               59.31
8     Vijayawada   9006240       5418303               60.16
9  Visakhapatnam  26308656      14854214               56.46

<p><p>

<p><p>

<p><p>

<u><h3> Promotion Type Analysis</h3><u>

**Promotion Types by Incremental Revenue**

In [32]:
promo_grp = data.groupby(['promo_type'])

sales_by_rev = pd.DataFrame(promo_grp['Incr_revenue'].sum().nlargest(5))

sales_by_rev

Incr_revenue
promo_type                
500 Cashback     122643000
BOGOF             69316990
33% OFF           15664212
50% OFF             709624
25% OFF           -1174864

**Promotion Types by Incremental Units Sold**

In [33]:
units_by_rev = pd.DataFrame(promo_grp['Incr_sold_units'].sum().nlargest(5))

units_by_rev

Incr_sold_units
promo_type                   
BOGOF                  157073
500 Cashback            40881
33% OFF                 27255
50% OFF                  6931
25% OFF                 -5717

**Average Percentage Revenue Increase on Products by Pomotion Type**

In [34]:
avg_perc_incr = pd.DataFrame(promo_grp['perc_incr_revenue'].mean().nlargest(5))

avg_perc_incr

perc_incr_revenue
promo_type                     
BOGOF                 261.22272
500 Cashback          162.17070
33% OFF                43.26685
50% OFF                33.48920
25% OFF               -15.40135

<p><p>

<p><p>

In [35]:
promo_grp = data.groupby(['promo_type'])
promo_tot_sales = pd.DataFrame(promo_grp['revenue'].sum().round(2)).reset_index()
promo_tot_sales.head()

promo_type    revenue
0       25% OFF    7998603
1       33% OFF   52204752
2       50% OFF    2872575
3  500 Cashback  189540000
4         BOGOF   95244220

In [36]:
promo_rev = pd.DataFrame(promo_grp['Incr_revenue'].sum()).reset_index()
promo_rev.head()

promo_type  Incr_revenue
0       25% OFF      -1174864
1       33% OFF      15664212
2       50% OFF        709624
3  500 Cashback     122643000
4         BOGOF      69316990

In [37]:
promo_overview = pd.merge(promo_tot_sales, promo_rev,on='promo_type')

promo_overview['%revenue_increment'] = (promo_overview['Incr_revenue']/promo_overview['revenue']) * 100

promo_overview = promo_overview.round(2)

promo_overview

promo_type    revenue  Incr_revenue  %revenue_increment
0       25% OFF    7998603      -1174864              -14.69
1       33% OFF   52204752      15664212               30.01
2       50% OFF    2872575        709624               24.70
3  500 Cashback  189540000     122643000               64.71
4         BOGOF   95244220      69316990               72.78

<p>The BUY ONE GET ONE FREE and Cashback promotions seem to be the most effective with that former racking up an 
average of 261% increase in sale and the former producing and average increase of 161%, It is clear that these 
two are better performers compared to the dscount based promotions.</p>

<p><p>

<p><p>

<u><h3> Product and Category Analysis</h3><u>

In [38]:
product_data = pd.read_csv(".\\dataset\\dim_products.csv")
product_data.head()

product_code                     product_name           category
0          P01           Atliq_Masoor_Dal (1KG)  Grocery & Staples
1          P02     Atliq_Sonamasuri_Rice (10KG)  Grocery & Staples
2          P03          Atliq_Suflower_Oil (1L)  Grocery & Staples
3          P04     Atliq_Farm_Chakki_Atta (1KG)  Grocery & Staples
4          P05  Atliq_Scrub_Sponge_For_Dishwash          Home Care

In [39]:
product_merged = data.merge(product_data, on='product_code')

product_merged.head()

event_id store_id  campaign_id product_code  base_price promo_type  \
0    7f650b  STCBE-2  CAMP_SAN_01          P11         190    50% OFF   
1  8.02E+96  STBLR-4  CAMP_DIW_01          P11         190    50% OFF   
2    02ebe9  STBLR-8  CAMP_DIW_01          P11         190    50% OFF   
3    def6ff  STMDU-0  CAMP_DIW_01          P11         190    50% OFF   
4    3ce076  STMLR-1  CAMP_DIW_01          P11         190    50% OFF   

   pre_promo_sales  post_promo_sales  Incr_sold_units  Incr_revenue  \
0               34                52               18          3420   
1               91               116               25          4750   
2               66                88               22          4180   
3               70                89               19          3610   
4               38                50               12          2280   

   perc_incr_units  perc_incr_revenue  revenue  \
0            52.94              52.94     9880   
1            27.47              27.47    22040   
2            33.33              33.33    16720   
3            27.14              27.14    16910   
4            31.58              31.58     9500   

                            product_name       category  
0  Atliq_Doodh_Kesar_Body_Lotion (200ML)  Personal Care  
1  Atliq_Doodh_Kesar_Body_Lotion (200ML)  Personal Care  
2  Atliq_Doodh_Kesar_Body_Lotion (200ML)  Personal Care  
3  Atliq_Doodh_Kesar_Body_Lotion (200ML)  Personal Care  
4  Atliq_Doodh_Kesar_Body_Lotion (200ML)  Personal Care

**Incremental Revenue by Product Category**

In [40]:
product_cat_grp = product_merged.groupby(['category'])

product_cat_top_rev = pd.DataFrame(product_cat_grp['Incr_revenue'].sum().reset_index())

product_cat_top_rev

category  Incr_revenue
0             Combo1     122643000
1  Grocery & Staples      42624108
2    Home Appliances      25150390
3          Home Care      16087190
4      Personal Care        654274

In [41]:
product_tot_sales = pd.DataFrame(product_cat_grp['revenue'].sum().reset_index())
product_tot_sales

category    revenue
0             Combo1  189540000
1  Grocery & Staples   95615960
2    Home Appliances   34633500
3          Home Care   24944175
4      Personal Care    3126515

**Category Overview**

In [42]:
cat_overview = pd.merge(product_tot_sales, product_cat_top_rev,on='category')

cat_overview['%revenue_increment'] = (cat_overview['Incr_revenue']/cat_overview['revenue']) * 100

cat_overview = cat_overview.round(2)

cat_overview

category    revenue  Incr_revenue  %revenue_increment
0             Combo1  189540000     122643000               64.71
1  Grocery & Staples   95615960      42624108               44.58
2    Home Appliances   34633500      25150390               72.62
3          Home Care   24944175      16087190               64.49
4      Personal Care    3126515        654274               20.93

In [43]:
prod_promo = pd.DataFrame(product_merged.groupby(['category', 'promo_type']).size())
prod_promo

0
category          promo_type       
Combo1            500 Cashback  100
Grocery & Staples 25% OFF       100
                  33% OFF       200
                  BOGOF         100
Home Appliances   BOGOF         200
Home Care         25% OFF       200
                  BOGOF         200
Personal Care     25% OFF       100
                  50% OFF       300

**Incremental Revenue by Product**

In [44]:
product_grp = product_merged.groupby(['product_code','product_name'])

product_top_rev = pd.DataFrame(product_grp['Incr_revenue'].sum().reset_index())

product_top_rev = product_top_rev.sort_values(by='Incr_revenue',ascending=False)

product_top_rev

product_code                               product_name  Incr_revenue
14          P15       Atliq_Home_Essential_8_Product_Combo     122643000
3           P04               Atliq_Farm_Chakki_Atta (1KG)      18248700
13          P14             Atliq_waterproof_Immersion_Rod      17561340
1           P02               Atliq_Sonamasuri_Rice (10KG)      13720440
7           P08                  Atliq_Double_Bedsheet_set      12917450
2           P03                    Atliq_Suflower_Oil (1L)       8711196
12          P13                Atliq_High_Glo_15W_LED_Bulb       7589050
6           P07                             Atliq_Curtains       3517500
0           P01                     Atliq_Masoor_Dal (1KG)       1943772
10          P11      Atliq_Doodh_Kesar_Body_Lotion (200ML)        335350
11          P12        Atliq_Lime_Cool_Bathing_Bar (125GM)        158844
9           P10    Atliq_Cream_Beauty_Bathing_Soap (125GM)         89520
8           P09  Atliq_Body_Milk_Nourishing_Lotion (120ML)         70560
4           P05            Atliq_Scrub_Sponge_For_Dishwash        -42735
5           P06            Atliq_Fusion_Container_Set_of_3       -305025

In [45]:
product_grp1 = data.groupby(['product_code'])
product_tot_sales = pd.DataFrame(product_grp1['revenue'].sum().round(2)).reset_index()
product_tot_sales.head()

product_code   revenue
0          P01   6422652
1          P02  45782100
2          P03  14310708
3          P04  29100500
4          P05    274175

**Poducts Overview**

In [46]:
product_overview = pd.merge(product_tot_sales, product_top_rev,on='product_code')

product_overview['%revenue_increment'] = (product_overview['Incr_revenue']/product_overview['revenue']) * 100

product_overview = product_overview.round(2)

product_overview

product_code    revenue                               product_name  \
0           P01    6422652                     Atliq_Masoor_Dal (1KG)   
1           P02   45782100               Atliq_Sonamasuri_Rice (10KG)   
2           P03   14310708                    Atliq_Suflower_Oil (1L)   
3           P04   29100500               Atliq_Farm_Chakki_Atta (1KG)   
4           P05     274175            Atliq_Scrub_Sponge_For_Dishwash   
5           P06    1855880            Atliq_Fusion_Container_Set_of_3   
6           P07    4895100                             Atliq_Curtains   
7           P08   17919020                  Atliq_Double_Bedsheet_set   
8           P09     671830  Atliq_Body_Milk_Nourishing_Lotion (120ML)   
9           P10     483145    Atliq_Cream_Beauty_Bathing_Soap (125GM)   
10          P11    1334180      Atliq_Doodh_Kesar_Body_Lotion (200ML)   
11          P12     637360        Atliq_Lime_Cool_Bathing_Bar (125GM)   
12          P13   10474800                Atliq_High_Glo_15W_LED_Bulb   
13          P14   24158700             Atliq_waterproof_Immersion_Rod   
14          P15  189540000       Atliq_Home_Essential_8_Product_Combo   

    Incr_revenue  %revenue_increment  
0        1943772               30.26  
1       13720440               29.97  
2        8711196               60.87  
3       18248700               62.71  
4         -42735              -15.59  
5        -305025              -16.44  
6        3517500               71.86  
7       12917450               72.09  
8          70560               10.50  
9          89520               18.53  
10        335350               25.14  
11        158844               24.92  
12       7589050               72.45  
13      17561340               72.69  
14     122643000               64.71

**Products Most Significantly Impacted By Promotions**

In [47]:
good_promo_tbl = product_overview[product_overview['%revenue_increment'] > 50]

good_promo_tbl.drop(['revenue','Incr_revenue','%revenue_increment'],inplace=True,axis=1)

good_promo_tbl

product_code                          product_name
2           P03               Atliq_Suflower_Oil (1L)
3           P04          Atliq_Farm_Chakki_Atta (1KG)
6           P07                        Atliq_Curtains
7           P08             Atliq_Double_Bedsheet_set
12          P13           Atliq_High_Glo_15W_LED_Bulb
13          P14        Atliq_waterproof_Immersion_Rod
14          P15  Atliq_Home_Essential_8_Product_Combo

**Poor Performing Products After Promotions**

In [48]:
poor_promo_tbl = product_overview[product_overview['%revenue_increment'] < 1]

poor_promo_tbl.drop(['revenue','Incr_revenue','%revenue_increment'],inplace=True,axis=1)

poor_promo_tbl

product_code                     product_name
4          P05  Atliq_Scrub_Sponge_For_Dishwash
5          P06  Atliq_Fusion_Container_Set_of_3

In [49]:
prod_promo_grp = data.groupby(['product_code', 'promo_type'])

product_promo = pd.DataFrame(prod_promo_grp['Incr_revenue'].sum())

product_promo

Incr_revenue
product_code promo_type                
P01          33% OFF            1943772
P02          33% OFF           13720440
P03          25% OFF            -274404
             BOGOF              8985600
P04          25% OFF            -497350
             BOGOF             18746050
P05          25% OFF             -42735
P06          25% OFF            -305025
P07          BOGOF              3517500
P08          BOGOF             12917450
P09          25% OFF             -42300
             50% OFF             112860
P10          25% OFF             -13050
             50% OFF             102570
P11          50% OFF             335350
P12          50% OFF             158844
P13          BOGOF              7589050
P14          BOGOF             17561340
P15          500 Cashback     122643000